[context,API] Functions for accessing configuration options

jrohel opened the pull request:
1. C wrapper for `libdnf::OptionBinds::Item` class methods `getPriority()`, `getValueString()`, `newString()`.
2.  Function `dnf_conf_add_setopt` for adding setopt.

Added:
```
gchar *dnf_conf_main_get_option(const gchar *name, enum DnfConfPriority *priority, GError ** error);
gboolean dnf_conf_main_set_option(const gchar *name, enum DnfConfPriority priority, const gchar *value, GError ** error);
gboolean dnf_conf_add_setopt(const gchar * key, enum DnfConfPriority priority, const gchar * value, GError ** error);
```

jrohel committed:
[context,API] Functions for accessing main/global configuration options

In fact, it is a C wrapper for libdnf::OptionBinds::Item class methods
getPriority(), getValueString(), newString().

Added:
gchar *dnf_conf_main_get_option(const gchar *name,
    enum DnfConfPriority *priority, GError ** error);

gboolean dnf_conf_main_set_option(const gchar *name,
    enum DnfConfPriority priority, const gchar *value, GError ** error);

Conan-Kudo commented on libdnf/dnf-types.h:
```diff
@@ -107,6 +107,7 @@ typedef enum {
         DNF_ERROR_CANNOT_WRITE_CACHE,                   /* Since: 0.7.0 */
         DNF_ERROR_NO_CAPABILITY,                        /* Since: 0.7.0 */
         DNF_ERROR_REMOVAL_OF_PROTECTED_PKG,             /* Since: 0.7.0 */
+        DNF_ERROR_UNKNOWN_OPTION,                       /* Since: 0.56.0 */
```
The docstring needs to be updated for this.

jrohel replied:
I added docstring for the new error code.

To be honest, I don't like this enum. Old explicitly assigned numbers look chaotic. And DNF_ERROR_LAST equals 55 (54 before), but DNF_ERROR_UNFINISHED_TRANSACTION equals 66. It look messy.
And we must be careful that the newly added error codes do not conflict with the existing explicitly defined ones.

Conan-Kudo replied:
I agree with you on this, and this is something we should fix for libdnf v5.

jrohel committed:
[context,API] Function for adding setopt

Sets the option value and priority. Supports also repositories options.
Repository(ies) can be specified with globs.

Added:
gboolean dnf_conf_add_setopt(const gchar * key,
    enum DnfConfPriority priority, const gchar * value, GError ** error);

key is option_name or repository_id.option_name

jrohel committed:
Version 0.56.0

Conan-Kudo commented:
@rh-atomic-bot r+
